WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -192,6 +192,47 @@ def register
     end
   end
 
+  def process_payment
+    competition = competition_from_params
+    if current_user
+      registrations = competition.registrations
+      registration = registrations.find_by_user_id(current_user.id)
+    else
+      flash[:danger] = "Please register for the competition first, and then pay the entry fee."
```

Comment:
I'm confused by this else block. This looks more like the person isn't logged in, rather than not registered for the competition.